pkg/virtctl/imageupload/imageupload.go

```diff
@@ -206,6 +214,25 @@ func parseArgs(args []string) error {
 	return nil
 }
 
+func normalizeVolumeMode(volumeMode string, blockVolume bool) (string, error) {
+	if volumeMode != "" {
+		fmt.Printf("--block-volume is deprecated and its value is overriding by --volume-mode\n")
+
+		if volumeMode != string(v1.PersistentVolumeBlock) && volumeMode != string(v1.PersistentVolumeFilesystem) {
+			return "", fmt.Errorf("can not use %s for volume mode, use either %s or %s", volumeMode, string(v1.PersistentVolumeBlock), string(v1.PersistentVolumeFilesystem))
+		}
+
+		return volumeMode, nil
+	}
+
+	if blockVolume && volumeMode == "" {
+		fmt.Printf("--block-volume is deprecated, use --volume-mode instead\n")
```

Comment:
you warn in two places, and now you have two different strings

Reply:
> I'm not a big fan of this fuzzy semantics. If someone is using the new flag, they should not use also the deprecated one.

It can't avoid to use the depreacted flag `block-volume` as it has default value, and it should not error when new flag is set. So the solution I give is telling user that the `block-volume` is deprecated and been overriding by `volume-mode`.

That why there have two strings:
-  When new flag is not set, `block-volume` is used only and just tell user it's deprecated.
-  When new flag is set, as `block-volume` has default value, tell user it's deprecated and been overriding by the new flag.